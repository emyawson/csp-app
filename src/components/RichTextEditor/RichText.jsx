import React from 'react';
import { documentToHtmlString } from '@contentful/rich-text-html-renderer';
import { BLOCKS, MARKS, INLINES } from '@contentful/rich-text-types';
import PropTypes from 'prop-types';
import './RichText.scss';

const RichTextEditor = ({ data, style }) => {
  const options = {
    renderMark: {
      [MARKS.BOLD]: (text) => `<b>${text}</b>`,
    },
    renderInlines: {
      [INLINES.HYPERLINK]: (node, next) => `<a href="${node.data.uri}"${node.data.uri.startsWith('https://mydomain.com') ? '' : ' target="_blank"'}>${next(node.content)}</a>`,
      [INLINES.HYPERLINK]: ({ node }, children) => `<a href=${node.data.uri} target='_self'>${children}</a>`,
    },
    renderNode: {
      [BLOCKS.HEADING_1]: (node, next) => `<h2>${next(node.content.value)}</h2>`,
      [BLOCKS.HEADING_2]: (node, children) => `<h3>${children}</h3>`,
      [BLOCKS.PARAGRAPH]: (node, next) => `<p>${next(node.content).replace(/\n/g, '<br/>')}</p>`,
      [BLOCKS.EMBEDDED_ASSET]: (node) => `<Img fluid="${node.data.target.fluid}" />`,
    },

  };

  const myHtml = documentToHtmlString(data, options);
  return (
    <div className={`richtext-editorstyle ${style || ''}`} dangerouslySetInnerHTML={{ __html: myHtml }} />
  );
};

RichTextEditor.propTypes = {
  data: PropTypes.object,
  style: PropTypes.string,
};

export default RichTextEditor;
