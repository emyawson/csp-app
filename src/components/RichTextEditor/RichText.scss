.richtext-editorstyle {
  > p {
    color: inherit;
  }

  > p > b {
    font-weight: bold;
  }

  > p > i {
    font-style: italic;
  }
}


.richtext-editorstyle.holidayHours {
  > p {
    color: inherit;
    font-size: 14px;
    line-height: 29px;
  }
}
