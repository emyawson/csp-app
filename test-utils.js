import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
