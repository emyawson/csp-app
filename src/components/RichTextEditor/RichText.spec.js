import React from 'react';
import RichTextEditor from './RichText.jsx';

import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const richTextDocument = {
    nodeType: 'document',
    data: {},
    content: [
      {
        nodeType: 'paragraph',
        data: {},
        content: [
          {
            nodeType: 'text',
            value: 'Hello',
            data: {},
            marks: [{ type: 'bold' }],
          },
          {
            nodeType: 'text',
            value: ' world!',
            data: {},
            marks: [{ type: 'italic' }],
          },
        ],
      },
    ],
  };

  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<RichTextEditor data={richTextDocument} />, combinedState);
};

test('renders <richTextDocument/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();
});
