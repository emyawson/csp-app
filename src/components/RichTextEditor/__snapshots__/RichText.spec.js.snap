// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <richTextDocument/> 1`] = `
<div>
  <div
    class="richtext-editorstyle "
  >
    <p>
      <b>
        Hello
      </b>
      <i>
         world!
      </i>
    </p>
  </div>
</div>
`;
